### PR TITLE
new things for nuxt

### PR DESCRIPTION
**Pull request by FunctionalMetatable**
Stuff for Nuxt.

Includes:
Login System
Auth Middleware
Header/Footer pages
Tutorial Component
Renderer Component <= TODO: Code Syntax Highlighting Library in vue & project embed. cc @CubeyTheCube
Markdown Tutorial
Prettier Workflow
Contributors Page <= TODO: Proper animation

...and more. Not sure if this is gonna get merged anyways but ok

You can view it at https://flagclicked-nuxtification.9gr.vercel.app/

### CODE REVIEW

File: nuxt.config.js [hunk 1]
```diff
@@ -4,37 +4,41 @@ export default {
    ** Doc: https://vue-meta.nuxtjs.org/api/#metainfo-properties
    */
   head: {
-    title: "Flag Clicked",
+    titleTemplate: (chunk) => {
+      return chunk ? `${chunk} - When Flag Clicked` : "When Flag Clicked";
+    },
```
Comment (TheColaber) [inline nuxt.config.js:9]: I never knew about this but it looks great... I would like it to say something `How to create a platformer on When Flag Clicked` if it was on a tutorial. That would be pretty cool. Are there also any for the meta stuff? It's fine if this is not in this PR we could open another issue for it.

File: nuxt.config.js [hunk 2]
```diff
@@ -51,5 +55,9 @@ export default {
    ** Plugins to load before mounting the App
    ** Doc: https://nuxtjs.org/docs/2.x/directory-structure/plugins
    */
-  plugins: []
+  plugins: [],
+
+  env: {
+    backendURL: "https://api.whenflagclicked.org",
+  },
```
Comment (TheColaber) [inline nuxt.config.js:62]: <3 - btw this url has not been created yet but I'll tell ryker to create it

File: pages/tutorials/_id/index.vue [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        <Tutorial :id="id" :content="content" />
+      </div>
+    </div>
+    <Footer />
+  </div>
+</template>
+<script>
+export default {
+  head() {
+    title: "Loading Tutorial"
```
Comment (TheColaber) [inline pages/tutorials/_id/index.vue:15]: TODO - I'll explain later

**Review by TheColaber — COMMENTED** [review]
Epic codes once again... I'll reveiw some more later and contribute a bit more once this and your other PR gets merged 👍

File: nuxt.config.js (hunk 1 above)
Comment (FunctionalMetatable) [reply nuxt.config.js:9]: i've been learning nuxt from ocular's source code

File: pages/tutorials/_id/index.vue (hunk 3 above)
Comment (FunctionalMetatable) [reply pages/tutorials/_id/index.vue]: heh i forgot return

**Comment by TheColaber** [comment]
NOTE: (you may already know this but) A middleware should be in the frontend nuxt that gets the token in the user's cookies and verifies that the user's token is valid. If not we should remove the cookie and therefore they will be logged out.

**Comment by TheColaber** [comment]
Hey btw how did you get this on vercel??? I've always tried to but it always fails lol

**Comment by TheColaber** [comment]
I would like an action where the bot creates a vercel link on every PR... maybe you or I can create an issue on this to discuss stuff.

**Comment by FunctionalMetatable** [comment]
> Hey btw how did you get this on vercel??? I've always tried to but it always fails lol

header kinda broken.
for me, i have a hobby account so i can deploy it. i think you cant as this is under an organization. i'll see what i can do

File: layouts/default.vue [hunk 4]
```diff
@@ -1,16 +1,11 @@
 <template>
   <div class="container">
-    <Navbar />
```
Comment (TheColaber) [inline layouts/default.vue:3]: Navbar was removed

File: pages/index.vue [hunk 5]
```diff
@@ -1,10 +1,13 @@
 <template>
-  <center>
-    <div>
-      <h2>This Week's Featured Tutorial</h2>
-      <h3>Tutorial Name - Username</h3>
-    </div>
-  </center>
+  <div class="container">
+    <Navbar />
```
Comment (TheColaber) [inline pages/index.vue:3]: Why only in index.vue? Why not on every page?

File: layouts/default.vue (hunk 4 above)
Comment (FunctionalMetatable) [reply layouts/default.vue:3]: will fix soon

**Comment by TheColaber** [comment]
@FunctionalMetatable I think we got vercel to work :tada: Ready to merge? Then I will move backend here...

**Comment by TheColaber** [comment]
(I do see a couple of issues with the css but I'll fix those later :+1:

**Comment by FunctionalMetatable** [comment]
> (I do see a couple of issues with the css but I'll fix those later 👍

the navbar's position set to absolute just breaks the page

**Comment by FunctionalMetatable** [comment]
done middleware

**Comment by FunctionalMetatable** [comment]
@TheColaber could you authorize the deployments?

**Comment by FunctionalMetatable** [comment]
Deploying perfectly for me.
@TheColaber please could you review?

**Comment by CubeyTheCube** [comment]
when this is deployed will I see this on flagclicked.vercel.app?

**Comment by FunctionalMetatable** [comment]
> when this is deployed will I see this on flagclicked.vercel.app?

https://flagclicked-nuxtification.9gr.vercel.app/

**Comment by FunctionalMetatable** [comment]
not sure what I should add to this pr

File: pages/index.vue (hunk 5 above)
Comment (FunctionalMetatable) [reply pages/index.vue]: done

**Comment by TheColaber** [comment]
@FunctionalMetatable sorry for the wait, I'm trying to figure out what we should do with the auth and stuff.

**Comment by TheColaber** [comment]
@FunctionalMetatable Take a look at https://github.com/nuxt-community/markdownit-module. Use it instead of the other markdown library that's currently in use. Our goal is to move away from our libraries that aren't nuxty and move to the nuxty ones.

**Comment by FunctionalMetatable** [comment]
> @FunctionalMetatable Take a look at https://github.com/nuxt-community/markdownit-module. Use it instead of the other markdown library that's currently in use. Our goal is to move away from our libraries that aren't nuxty and move to the nuxty ones.

Marked works fine. It's an npm module, so it can be bundled client-side as well. With marked, we can render scratchblocks, add our own flavored markdown, etc. With the _nuxty_ ones we cannot actually implement our own flavored markdown with

**Comment by markverb1** [comment]
> > @FunctionalMetatable Take a look at https://github.com/nuxt-community/markdownit-module. Use it instead of the other markdown library that's currently in use. Our goal is to move away from our libraries that aren't nuxty and move to the nuxty ones.
> 
> Marked works fine. It's an npm module, so it can be bundled client-side as well. With marked, we can render scratchblocks, add our own flavored markdown, etc. ~With the _nuxty_ ones we cannot actually implement our own flavored markdown with~
> Edit: I'll check the markdown-it plugins thing

i think you can add plugins or whatever you call them
###### https://stackoverflow.com/questions/59680780/creating-a-custom-markdown-it-plugin-for-nuxt-application-results-in-typeerror-w

**Comment by FunctionalMetatable** [comment]
I replaced `marked` with `markdown-it` and added Syntax highlighting. Could this get merged/reviewed? (I personally **don't** want to be doing the whole nuxtification)

**Comment by TheColaber** [comment]
Thanks for your hard work, I'll merge this and work on a couple things later.

**Pull request merged**